In [2]:
import numpy as np
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.model_selection import cross_val_score
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
from sklearn.datasets import load_iris

In [4]:
iris=load_iris()

#LogisticRegrssion

In [6]:
lr_scores = cross_val_score(LogisticRegression(max_iter=1000), iris.data, iris.target)
lr_scores

array([0.96666667, 1.        , 0.93333333, 0.96666667, 1.        ])

In [8]:
np.average(lr_scores)

0.9733333333333334

#DecisionTreeClassifier

In [9]:
dt_scores = cross_val_score(DecisionTreeClassifier(), iris.data, iris.target)
dt_scores

array([0.96666667, 0.96666667, 0.9       , 0.96666667, 1.        ])

In [10]:
np.average(dt_scores)

0.9600000000000002

#SVM

In [11]:
svm_scores = cross_val_score(SVC(), iris.data, iris.target)
svm_scores

array([0.96666667, 0.96666667, 0.96666667, 0.93333333, 1.        ])

In [13]:
np.average(svm_scores)

0.9666666666666666

#RandomForestClassifier

In [23]:
rf_scores = cross_val_score(RandomForestClassifier(n_estimators=30), iris.data, iris.target)
rf_scores

array([0.96666667, 0.96666667, 0.93333333, 0.96666667, 1.        ])

In [24]:
np.average(rf_scores)

0.9666666666666668

**Here above the Best score so far is from LogisticRegression: 0.9733333333333334**

#Instead of finding best score individually, we can make a function in this way:

In [25]:
import numpy as np
from sklearn.model_selection import cross_val_score
from sklearn.datasets import load_iris
from sklearn.linear_model import LogisticRegression
from sklearn.tree import DecisionTreeClassifier
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier

# Load the iris dataset
iris = load_iris()

# Define a list of classifiers
classifiers = [
    LogisticRegression(max_iter=1000),
    DecisionTreeClassifier(),
    SVC(),
    RandomForestClassifier(n_estimators=30)
]

# Function to calculate average score for a classifier
def get_avg_score(classifier, data, target):
    scores = cross_val_score(classifier, data, target)
    return np.average(scores)

# Calculate average scores for each classifier
avg_scores = [get_avg_score(clf, iris.data, iris.target) for clf in classifiers]

# Find the best score and corresponding classifier index
best_score_index = np.argmax(avg_scores)
best_score = avg_scores[best_score_index]
best_classifier = classifiers[best_score_index]

# Print the best score and corresponding classifier
print(f"Best average score: {best_score}")
print(f"Corresponding classifier: {best_classifier}")


Best average score: 0.9733333333333334
Corresponding classifier: LogisticRegression(max_iter=1000)
